lib/plexy/instrumentor.ex

```diff
@@ -34,6 +34,8 @@ defmodule Plexy.Instrumentor do
       stop = :erlang.monotonic_time()
       diff = :erlang.convert_time_unit(stop - start, :native, :milli_seconds)
 
+      Logger.measure("request.latency.ms", diff)
```

Comment:
That also matches [the examples in the Dev Center article](https://devcenter.heroku.com/articles/librato#custom-distributions).

Reply:
FWIW: The internal docs specifically say to include the unit in the metric name. There are plenty of services that do this internally because they follow our standards.

I remember arguing with Joe when he wrote that Devcenter article, but I understand *why* they did what they did, they just never ended up completing implementations of how to deal with metric changes (and TBH I don't think they ever will).

In summary though here is why you should include the unit in the name:

Including the measurement in the name reduces ambiguity for users of the metric. The unit is present in basically every context the metric can be used in librato. This isn't the case when it's specified after the value, it's hidden until the metric makes it's way to a chart or the attributes are viewed. Additionally changing the unit in `measure#foo.bar=200ms` outputs doesn't update the until in librato, it will still end up being whatever unit was originally encountered.

If the unit of a measure gets changed it creates a new metric. Changing the unit of a measure fundamentally changes the nature of the derived metrics. If you think through what happens when the reporting unit changes, if the metric stays the same (foo.bar=200ms => foo.bar=0.2s; metric == foo.bar) then how are rollups, which were reported in the old unit, utilized?

It's easier to extract metrics in other downstream systems (ie splunk) when the unit is part of the measure name vs when it's part of the value.

Reply:
My counterargument would be that units are unlikely to change (deciding to always favor milliseconds would be better) and that it's arguably worse if they do change without that change being reflected in the metric name.

Your explanation makes sense to me, though.